tck/utils/key.ts

```diff
@@ -0,0 +1,104 @@
+import { PrivateKey, PublicKey, KeyList, Key } from "@hashgraph/sdk";
+import { proto } from "@hashgraph/proto";
+
+import { asn1DecodeStringDer } from "../utils/asn1-decoder";
+
+const getKeyFromString = (
+    keyString: string,
+): PublicKey | PrivateKey | Key | null => {
+    const decodedKey = asn1DecodeStringDer(keyString);
+
+    if (decodedKey.isKeyListHex) {
+        return getKeyFromKeyListHex(keyString);
+    }
+
+    try {
+        if (decodedKey.isPublicKey) {
+            if (decodedKey.keyTypes.includes("ecdsa")) {
+                return PublicKey.fromStringECDSA(keyString);
+            } else if (decodedKey.keyTypes.includes("ed25519")) {
+                return PublicKey.fromStringED25519(keyString);
+            }
+        } else {
+            if (decodedKey.keyTypes.includes("ecdsa")) {
+                return PrivateKey.fromStringECDSA(keyString);
+            } else if (decodedKey.keyTypes.includes("ed25519")) {
+                return PrivateKey.fromStringED25519(keyString);
+            }
+        }
+    } catch (error) {
+        console.error("Error parsing key: ", error);
+        return null;
+    }
+
+    return null;
+};
+
+const getKeyFromKeyListHex = (keyString: string): KeyList => {
+    try {
+        const byteArray = Buffer.from(keyString, "hex");
+
+        let decodedKeyList: proto.KeyList;
+        let decodedThresholdKey: proto.ThresholdKey;
+
+        // Attempt to decode as KeyList
+        try {
+            decodedKeyList = proto.KeyList.decode(byteArray);
+        } catch (keyListError) {
+            // Expected that this may fail if it's not a KeyList
```

Comment:
I think you can do `console.error` here?

Reply:
Since it's expected, there's no need for this. Please refer to the previous comment for the rationale.